gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java

```diff
@@ -215,6 +210,91 @@ public static List<CommentStatement> createRpcMethodHeaderComment(
+    String ASYNC_METHODS =
+        "<p>Methods that return long-running operations have \"Async\" method variants that return `OperationFuture`, which is used to track polling of the service.</p>\n";
+
+    StringBuilder tableBuilder = new StringBuilder();
+    tableBuilder
+        .append("<table>\n")
+        .append("   <tr>\n")
+        .append("     <th>Method</th>\n")
+        .append("     <th>Description</th>\n")
+        .append("     <th>Method Variants</th>\n");
+    for (MethodAndVariants method : methodAndVariantsList) {
+      tableBuilder
+          .append("   <tr>\n")
+          .append("     <td>")
+          .append(method.method)
+          .append("</td>\n")
+          .append("     <td>")
+          .append("<p>" + method.description + "</p>")
+          .append("</td>\n")
+          .append("     <td>\n");
+      generateUnorderedListMethodVariants(
+          tableBuilder, REQUEST_OBJECT_METHODS, method.requestObjectVariants);
+      generateUnorderedListMethodVariants(
+          tableBuilder, FLATTENED_METHODS, method.flattenedVariants);
+      generateUnorderedListMethodVariants(tableBuilder, ASYNC_METHODS, method.asyncVariants);
+      generateUnorderedListMethodVariants(tableBuilder, CALLABLE_METHODS, method.callableVariants);
+      tableBuilder.append("   </td>\n").append("   </tr>\n");
+    }
+    tableBuilder.append(" </table>\n");
+    return tableBuilder.toString();
+  }
+
+  private static void generateUnorderedListMethodVariants(
+      StringBuilder tableBuilder, String methodType, List<String> methodVariants) {
+    if (!methodVariants.isEmpty()) {
+      tableBuilder
+          .append("     " + methodType + "     ")
+          .append("<ul>\n")
+          .append("     <li>")
+          .append(String.join("\n     <li>", methodVariants))
+          .append("\n")
+          .append("     </ul>")
+          .append("\n");
+    }
+  }
+
+  private static class MethodAndVariants {
+    private final String method;
+    private final String description;
+
+    private final List<String> flattenedVariants;
+    private final List<String> requestObjectVariants;
+    private final List<String> callableVariants;
+    private final List<String> asyncVariants;
+
+    private MethodAndVariants(String method, String description, List<String> methodVariants) {
+      this.method = method;
+      this.description = description;
+      requestObjectVariants =
+          methodVariants.stream().filter(s -> s.contains("request")).collect(toList());
+      methodVariants.removeAll(requestObjectVariants);
+      callableVariants =
+          methodVariants.stream().filter(s -> s.contains("Callable")).collect(toList());
+      methodVariants.removeAll(callableVariants);
+      asyncVariants = methodVariants.stream().filter(s -> s.contains("Async")).collect(toList());
+      methodVariants.removeAll(asyncVariants);
+      flattenedVariants = methodVariants;
```

Comment:
Is there no way to search for the flattenedVariants besides ruling out the others? If not, could you add a comment for that.

Reply:
Added a comment - flattenedVariants don't have a suffix appended, so the easiest way is to rule out the others.